test/funcfile-docker-rt-tests/fn_test.go

```diff
@@ -0,0 +1,165 @@
+package tests
+
+import (
+        "bytes"
+	"fmt"
+	"io/ioutil"
+        "log"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+)
+
+
+
+func TestDockerRuntime(t *testing.T) {
+        testname := "TestDockerRuntime"
+        testfiles := []string{ "Dockerfile", "func.go" }
+
+        currdir, err := os.Getwd() 
+	if err != nil {
+		t.Fatalf("ERROR: Failed to get current directory: err: %s", err)
+	}
+	testdir, err := ioutil.TempDir("", "cli_test_funcfile-docker-rt_" + testname)
+	if err != nil {
+		t.Fatalf("ERROR: Failed to make tmp test directory: err: %s", err)
+	}
+        defer cleanup(t, currdir, testdir)
+	fnTestBin := setupTestFiles(t, testname, currdir, testdir, testfiles)   
+	runFnInit(t, testname, fnTestBin)
+	runFnBuild(t, testname, fnTestBin)
+	runFnRun(t, testname, fnTestBin)
+}
+
+func TestDockerRuntimeNoDockerfile(t *testing.T) {
+        testname := "TestDockerRuntimeNoDockerfile"
+        testfiles := []string{ "func.yaml", "func.go" }
+
+        currdir, err := os.Getwd() 
+	if err != nil {
+		t.Fatalf("ERROR: Failed to get current directory: err: %s", err)
+	}
+	testdir, err := ioutil.TempDir("", "cli_test_funcfile-docker-rt_" + testname)
+	if err != nil {
+		t.Fatalf("ERROR: Failed to make tmp test directory: err: %s", err)
+	}
+        defer cleanup(t, currdir, testdir)
+	fnTestBin := setupTestFiles(t, testname, currdir, testdir, testfiles)   
+	runFnBuildNoDockerfile(t, testname, fnTestBin)
+}
+
+
+func cleanup(t *testing.T, currdir, testdir string) {
+        err := os.Chdir(currdir);
+	if err != nil {
+		t.Fatalf("ERROR: Failed to cd " + currdir + " directory: err: %s", err)
+	}
+	os.Remove(testdir);
+}
+
+
+func setupTestFiles(t *testing.T, 
+	testname, currdir, testdir string, testfiles []string) string {
+
+        mylog("INFO", testname, "Current directory is " + currdir)
+
+        testfilesdir := path.Join(currdir, "testfiles")
+        for _, testfile :=range testfiles {
+	        err := copyFile(path.Join(testfilesdir, testfile), path.Join(testdir, testfile))
+		if err != nil {
+			t.Fatalf("ERROR: Failed to copy " + testfile + " to test directory: err: %s", err)
+		}
+	}
+
+	fnTestBin := path.Join(testdir, "fn")
+        err := os.Chdir("../../")
+	if err != nil {
+		t.Fatalf("ERROR: Failed to cd ../../ directory: err: %s", err)
```

Comment:
err.Error()

Reply:
changed to use %v to format 'err' in the test